app/Services/SignService.php

```diff
@@ -38,37 +39,17 @@ public function editSign(Request $request, $user): Sign
     }
 
     /**
-     * @param Request $request
-     * @param $user
-     * @return Sign|\Illuminate\Database\Eloquent\Model
+     * @param  SignRequest  $request
+     * @param  User  $user
+     * @return Sign|Model
      */
-    private function newSign(Request $request, User $user): Sign
+    private function newSign(SignRequest $request, User $user): Sign
     {
-        $this->validateSign($request);
-
         $language = $this->languageRepository->getSigned();
-        $sign = $this->signRepository->make(
-            $request->input('video_uuid'),
-            $request->input('video_url'),
-            $request->input('thumbnail_url'),
-            $request->input('small_thumbnail_url'),
-            $language,
-            $user
-        );
+        $sign = $this->signRepository->make($request, $language, $user);
```

Comment:
Oh, og den smarte ved Form Request objecter er at de allerede er "clean". Så hvis du kan udfylde Request'en på en god måde allerede fra EndPoints (At du også tilføjer creator og editor og de andre variabler ind, og validere dem) - kan du kalde Sign::make($request), og vupti, så er hele modellen oprettet uden at du behøver at gøre rede på de forskellige variabler. Det er i hvert fald en idé her... Læs mere på https://medium.com/damilola-jolayemi/making-the-most-of-laravel-form-requests-8f07767859fc

Reply:
Som udgangspunkt ser jeg helst at al kommunikation med databasen skal ske i repository. Samtidigt er det lidt af blackbox når man bare bruge make($request). Jeg er stadig fan af at kunne se hvad hver input består af.

Reply:
Approved